plugins/modules/ndo_tenant_custom_qos_policy.py

```diff
@@ -0,0 +1,548 @@
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate, KVPair
+from ansible_collections.cisco.mso.plugins.module_utils.utils import append_update_ops_data
+from ansible_collections.cisco.mso.plugins.module_utils.constants import (
+    TARGET_DSCP_MAP,
+    TARGET_COS_MAP,
+    DSCP_COS_KEY_MAP,
+    QOS_LEVEL,
+)
+
+
+def main():
+    COS_MAP_CHOICES = list(TARGET_COS_MAP) + list(TARGET_COS_MAP.values())
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True, aliases=["tenant_template"]),
+        name=dict(type="str", aliases=["custom_qos_policy"]),
+        uuid=dict(type="str", aliases=["custom_qos_policy_uuid"]),
+        description=dict(type="str"),
+        dscp_mappings=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                dscp_from=dict(type="str", choices=list(TARGET_DSCP_MAP), aliases=["from"]),
+                dscp_to=dict(type="str", choices=list(TARGET_DSCP_MAP), aliases=["to"]),
+                dscp_target=dict(type="str", choices=list(TARGET_DSCP_MAP), aliases=["target"]),
+                target_cos=dict(type="str", choices=COS_MAP_CHOICES),
+                qos_priority=dict(
+                    type="str",
+                    choices=QOS_LEVEL,
+                    aliases=["priority", "prio"],
+                ),
+            ),
+        ),
+        cos_mappings=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                dot1p_from=dict(type="str", choices=COS_MAP_CHOICES, aliases=["from"]),
+                dot1p_to=dict(type="str", choices=COS_MAP_CHOICES, aliases=["to"]),
+                dscp_target=dict(type="str", choices=list(TARGET_DSCP_MAP), aliases=["target"]),
+                target_cos=dict(type="str", choices=COS_MAP_CHOICES),
+                qos_priority=dict(
+                    type="str",
+                    choices=QOS_LEVEL,
+                    aliases=["priority", "prio"],
+                ),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    dscp_mappings = module.params.get("dscp_mappings")
+    if isinstance(dscp_mappings, list):
```

Comment:
Do we need the list check, think the value can only be None or a List. Think we only need to jump to comprehension logic when the value is not falsy. Is below enough?

```suggestion
    if dscp_mappings:
```

Reply:
I think it is, thanks!